hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -236,19 +237,20 @@
   public static final String HBASE_MAX_CELL_SIZE_KEY = "hbase.server.keyvalue.maxsize";
   public static final int DEFAULT_MAX_CELL_SIZE = 10485760;
 
-  /**
-   * This is the global default value for durability. All tables/mutations not
-   * defining a durability or using USE_DEFAULT will default to this value.
-   */
-  private static final Durability DEFAULT_DURABILITY = Durability.SYNC_WAL;
-
   public static final String HBASE_REGIONSERVER_MINIBATCH_SIZE =
       "hbase.regionserver.minibatch.size";
   public static final int DEFAULT_HBASE_REGIONSERVER_MINIBATCH_SIZE = 20000;
 
   public static final String WAL_HSYNC_CONF_KEY = "hbase.wal.hsync";
   public static final boolean DEFAULT_WAL_HSYNC = false;
 
+  /**
+   * This is for for using HRegion as a local storage, where we may put the recovered edits in a
+   * special place. Once this is set, we will only replay the recovered edits under this directory
+   * and ignore the original replay directory configs.
+   */
+  public static final String RECOVERED_EDITS_DIR = "hbase.hregion.recovered.edits.dir";
```

Comment:
Why add this? Why donnot use recovered.edits dir? And if we add this config, it should use which filesystem? root or wal?

Reply:
We have some assumptions for the recovered edits files, for example the name, but as you commented above, now we just move the wal file to the 'recovered edits' directory, we will break the assumption. So I modified the code in replayRecoveredEdits a bit that, if this config is set, we will just replay the files under this directory one by one. And it should be on the wal file system.